domain/consensus/utils/consensushashing/calculate_signature_hash_test.go

```diff
@@ -276,3 +279,113 @@ func generateTxs() (nativeTx, subnetworkTx *externalapi.DomainTransaction, err e
 
 	return nativeTx, subnetworkTx, nil
 }
+
+func BenchmarkCalculateSignatureHash(b *testing.B) {
+	sigHashTypes := []consensushashing.SigHashType{
+		consensushashing.SigHashAll,
+		consensushashing.SigHashNone,
+		consensushashing.SigHashSingle,
+		consensushashing.SigHashAll | consensushashing.SigHashAnyOneCanPay,
+		consensushashing.SigHashNone | consensushashing.SigHashAnyOneCanPay,
+		consensushashing.SigHashSingle | consensushashing.SigHashAnyOneCanPay}
+
+	for _, size := range []int{10, 100, 1000} {
+		tx := generateTransaction(b, sigHashTypes, size)
+
+		b.Run(fmt.Sprintf("%d-inputs-and-outputs", size), func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				reusedValues := &consensushashing.SighashReusedValues{}
+				sigHashType := sigHashTypes[i%len(sigHashTypes)]
```

Comment:
It's best to not use the benchmark index in the benchmark, because it scales it up without taking divisable by `len(sigHashTypes)` into account so the result will be inconsistent between scales

Reply:
That's actually a bug, I planned to use inputIndex, not i